internal/hcl/parser.go

```diff
@@ -397,9 +397,31 @@ func (p *Parser) ParseDirectory() (m *Module, err error) {
 		nil,
 	)
 
-	root, err := evaluator.Run()
-	if err != nil {
-		return m, err
+	var root *Module
+
+	// Graph evaluation
+	if os.Getenv("INFRACOST_GRAPH_EVALUATOR") == "true" {
+		g, err := NewGraphWithRoot(p.logger)
+		if err != nil {
+			return m, err
+		}
+
+		err = g.Populate(evaluator)
+		if err != nil {
+			return m, err
+		}
+
+		g.ReduceTransitively()
+		g.Walk()
+		evaluator.module.Blocks = evaluator.filteredBlocks
+		evaluator.module = *evaluator.collectModules()
+		root = &evaluator.module
```

Comment:
I'd prefer if we have a equivalent interface to the current evaluator so it's easier to understand that the `Graph` and `Evaluator` and equivalent in what they are doing. Somthing like:

```suggestion
		root, err := g.Run(evaluator)
		if err != nil {
			return m, err
		}
```